### PR TITLE
feat(api): add routines to handle lost run/jobs/workers

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/v2_workflow_run_job_routines.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+	ctx = context.WithValue(ctx, cdslog.Workflow, runJob.WorkflowName)
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback() // nolint
+
+	log.Info(ctx, fmt.Sprintf("reEnqueueScheduledJob: re-enqueue job %s/%s on workflow %s run %d", runJob.JobID, runJob.ID, runJob.WorkflowName, runJob.RunNumber))
```
Comment (fsamin) [inline engine/api/v2_workflow_run_job_routines.go:109]: Add schedule time in the log

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19067.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19069.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: engine/worker/internal/takeV2.go [hunk 2]
```diff
@@ -81,6 +81,52 @@ func (w *CurrentWorker) V2Take(ctx context.Context, region, jobRunID string) err
...
 	w.SetGelfLogger(h, l)
 
+	//This goroutine try to get the job every 5 seconds, if it fails, it cancel the build.
+	tick := time.NewTicker(5 * time.Second)
+	go func(cancel context.CancelFunc, runJobID string, tick *time.Ticker) {
+		var nbConnrefused int
+		for {
+			select {
+			case <-ctx.Done():
+				return
```
Comment (fsamin) [inline engine/worker/internal/takeV2.go:91]: stop the ticker

**Review by fsamin — CHANGES_REQUESTED** [review]

File: engine/worker/internal/takeV2.go (hunk 2 above)
Comment (sguiheux) [reply engine/worker/internal/takeV2.go:92]: done

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19071.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Pull request merged**